src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,122 @@
-import { useState } from 'react';
+/* eslint-disable max-len */
+import { SyntheticEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const isDisabled = !title || !imgUrl || !imdbUrl || !imdbId;
```

Comment:
```suggestion
  const isDisabled = !title.trim() || !imgUrl || !imdbUrl || !imdbId;
```
it's better to trim text before such checking, I can add the new movie with only spaces in inputs:

![image](https://user-images.githubusercontent.com/86923089/226170538-e6f0ad33-aea7-4fec-a734-92df20ed9db7.png)

Just FYI